internal/internal.go

```diff
@@ -492,35 +493,57 @@ func getResponseStatus(result map[string]interface{}) (int32, error) {
 func getResponseHeaders(result map[string]interface{}) ([]*ext_core.HeaderValueOption, error) {
 	var ok bool
 	var val interface{}
-	var headers map[string]interface{}
 
 	responseHeaders := []*ext_core.HeaderValueOption{}
 
 	if val, ok = result["headers"]; !ok {
 		return responseHeaders, nil
 	}
 
-	if headers, ok = val.(map[string]interface{}); !ok {
-		return nil, fmt.Errorf("type assertion error")
-	}
-
-	for key, value := range headers {
-		var headerVal string
-		if headerVal, ok = value.(string); !ok {
-			return nil, fmt.Errorf("type assertion error")
+	takeResponseHeaders := func(headers map[string]interface{}) ([]*ext_core.HeaderValueOption, error) {
+		responseHeaders := []*ext_core.HeaderValueOption{}
+		for key, value := range headers {
+			var headerVal string
+			if headerVal, ok = value.(string); !ok {
+				return nil, fmt.Errorf("type assertion error")
+			}
+			headerValue := &ext_core.HeaderValue{
+				Key:   key,
+				Value: headerVal,
+			}
+			headerValueOption := &ext_core.HeaderValueOption{
+				Header: headerValue,
+			}
+			responseHeaders = append(responseHeaders, headerValueOption)
 		}
+		return responseHeaders, nil
+	}
 
-		headerValue := &ext_core.HeaderValue{
-			Key:   key,
-			Value: headerVal,
+	switch val.(type) {
+	case []interface{}:
+		for _, vval := range val.([]interface{}) {
+			if headers, ok := vval.(map[string]interface{}); ok {
+				responseHeadersToAppend, err := takeResponseHeaders(headers)
+				if err != nil {
+					return nil, err
+				}
+				responseHeaders = append(responseHeaders, responseHeadersToAppend...)
+			} else {
+				return nil, fmt.Errorf("type assertion error")
+			}
```

Comment:
We could try to avoid a level of indentation here

```suggestion
			headers, ok := vval.(map[string]interface{})
			if !ok {
				return nil, fmt.Errorf("type assertion error")
			}
			
			responseHeadersToAppend, err := takeResponseHeaders(headers)
			if err != nil {
				return nil, err
			}
			responseHeaders = append(responseHeaders, responseHeadersToAppend...)
```

It's a matter of taste, I suppose. 💭

Reply:
Thanks for adding this @keyolk ! It would be useful if you could add a policy example in [this](https://github.com/open-policy-agent/opa-envoy-plugin#example-policy-with-object-response) section with some text (like Stephan mentioned) and also add a policy example.